### PR TITLE
ratelimit support stat prefix

### PR DESCRIPTION
**Pull request by StarryVae**
Commit Message: ratelimit support stat prefix
Additional Description: N/A
Risk Level: low
Testing: ut
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by StarryVae** [comment]
issue #27421

File: api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto [hunk 1]
```diff
@@ -130,6 +130,10 @@ message RateLimit {
   // Sets the HTTP status that is returned to the client when the ratelimit server returns an error
   // or cannot be reached. The default status is 500.
   type.v3.HttpStatus status_on_error = 12;
+
+  // Optional additional prefix to use when emitting statistics. This allows to distinguish
+  // emitted statistics between configured ``ratelimit`` filters in an HTTP filter chain.
+  string stat_prefix = 13;
```
Comment (htuch) [inline api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto:136]: Should update ratelimit stats docs as well: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rate_limit_filter.html#statistics

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo nit.

File: api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto (hunk 1 above)
Comment (StarryVae) [reply api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto:136]: done, ptal again, thanks.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by StarryVae** [comment]
the ci seems to be blocked by flake coverage #30701

**Comment by htuch** [comment]
@StarryVae can you rebase and push again? Thanks.

**Comment by StarryVae** [comment]
done, thanks. @htuch

**Review by htuch — APPROVED** [review]

**Pull request merged**